File: CH8/startwithvuejs/src/services/FoodServiceService.ts
import Vue from 'vue'
import { FoodService } from '../entities/FoodService'
import { User } from '../entities/User'
import { Image } from '../entities/Image'

export class FoodServiceService {
  static getById (email: string) {
    return Vue.axios.get<FoodService>(`/foodservices/${email}`)
  }

  static create (foodService: FoodService) {
    let formData = new FormData()

    formData.append('file', foodService.image)

    return Vue.axios.post<Image>(`/files`, formData,
      {
        headers: {
          'Content-Type': 'multipart/form-data'
        }
      })
      .then(response => {
        foodService.imageUrl = response.data.imageUrl

        return Vue.axios.post<FoodService>(`/foodservices`, foodService)
      })
  }

  static update (foodService: FoodService) {
    let formData = new FormData()

    formData.append('file', foodService.image)

    return Vue.axios.post<Image>(`/files`, formData,
      {
        headers: {
          'Content-Type': 'multipart/form-data'
        }
      })
      .then(response => {
        foodService.imageUrl = response.data.imageUrl

        return Vue.axios.put<FoodService>(`/foodservices`, foodService)
      })
  }

  static login (user: User) {
    return Vue.axios.post<FoodService>(`/foodservices/login`, user)
  }

  static getByFoodType (foodType:string, page:number, pageSize:number) {
    return Vue.axios.get<Array<FoodService>>(`/foodservices?foodType=${foodType}&page=${page}&pageSize=${pageSize}`)
  }
}

File: CH8/StartWithJEE/StartWithJEE-ejb/src/main/java/com/daniel/delivery/abstraction/service/foodservice/FoodServiceService.java

package com.daniel.delivery.abstraction.service.foodservice;

import com.daniel.delivery.abstraction.entity.FoodService;
import com.daniel.delivery.abstraction.entity.User;
import java.util.List;
import java.util.Optional;

public interface FoodServiceService {
    Optional<FoodService> login(User user);
    Optional<FoodService> getById(String email);
    FoodService save(FoodService foodService);
    FoodService update(FoodService foodService);
    FoodService deActivate(String email);
    List<FoodService> getByFoodType(String foodType, Integer page, Integer pageSize);
}
